test/modeling_library/distributions.jl

```diff
@@ -233,6 +233,34 @@ end
     @test isapprox(actual[3][2, 2], finite_diff_mat_sym(f, args, 3, 2, 2, dx))
 end
 
+@testset "dirichlet" begin
+    x = dirichlet([1., 1., 1., 1.])
+    @test length(x) == 4
+    @test isapprox(sum(x), 1.)
+
+    # bounds checking
+    @test logpdf(dirichlet, [0., 0], [1., 1.]) == -Inf
+    @test logpdf(dirichlet, [1., 1.], [1., 1.]) == -Inf
+    @test logpdf(dirichlet, [2., -1], [1., 1.]) == -Inf
+    @test logpdf(dirichlet, [.5, .5], [-1., 1.]) == -Inf
+    @test logpdf(dirichlet, [.5, .5], [-1., 1.]) == -Inf
+    @test logpdf(dirichlet, [0., 1], [1., 1.]) != -Inf
+
+    @test isapprox(logpdf(dirichlet, [.01, .99], [2., 2.]),
+                   Distributions.logpdf(Distributions.Dirichlet([2., 2.]), [.01, .99]))
+    @test isapprox(logpdf(dirichlet, [.01, .99], [1., 4.]),
+                   Distributions.logpdf(Distributions.Dirichlet([1., 4.]), [.01, .99]))
+    @test isapprox(logpdf(dirichlet, [.01, .99], [.01, .01]),
+                   Distributions.logpdf(Distributions.Dirichlet([.01, .01]), [.01, .99]))
```

Comment:
Can we add tests for Dirichlet distributions over 3 categories as well? Just to make sure it works beyond the case where it's equivalent to a Beta distribution.